src/main/kotlin/no/nav/familie/ef/sak/task/BehandlingsstatistikkTask.kt

```diff
@@ -0,0 +1,151 @@
+package no.nav.familie.ef.sak.task
+
+import no.nav.familie.kontrakter.felles.objectMapper
+import java.time.LocalDateTime
+import com.fasterxml.jackson.module.kotlin.readValue
+import no.nav.familie.ef.sak.api.beregning.ResultatType
+import no.nav.familie.ef.sak.api.beregning.VedtakService
+import no.nav.familie.ef.sak.integration.dto.pdl.AdressebeskyttelseGradering
+import no.nav.familie.ef.sak.integration.dto.pdl.gjeldende
+import no.nav.familie.ef.sak.iverksett.IverksettClient
+import no.nav.familie.ef.sak.repository.domain.Fagsak
+import no.nav.familie.ef.sak.repository.domain.Stønadstype
+import no.nav.familie.ef.sak.service.BehandlingService
+import no.nav.familie.ef.sak.service.FagsakService
+import no.nav.familie.ef.sak.service.OppgaveService
+import no.nav.familie.ef.sak.service.PersonService
+import no.nav.familie.ef.sak.service.SøknadService
+import no.nav.familie.kontrakter.ef.felles.BehandlingType
+import no.nav.familie.kontrakter.ef.felles.StønadType
+import no.nav.familie.kontrakter.ef.iverksett.BehandlingsstatistikkDto
+import no.nav.familie.kontrakter.ef.iverksett.Hendelse
+import no.nav.familie.kontrakter.felles.oppgave.Oppgave
+import no.nav.familie.prosessering.AsyncTaskStep
+import no.nav.familie.prosessering.TaskStepBeskrivelse
+import no.nav.familie.prosessering.domene.Task
+import org.springframework.stereotype.Service
+import java.time.ZoneId
+import java.util.Properties
+import java.util.UUID
+
+@Service
+@TaskStepBeskrivelse(
+        taskStepType = BehandlingsstatistikkTask.TYPE,
+        beskrivelse = "Sender behandlingsstatistikk til iverksett"
+)
+
+class BehandlingsstatistikkTask(private val iverksettClient: IverksettClient,
+                                private val behandlingService: BehandlingService,
+                                private val fagsakService: FagsakService,
+                                private val søknadService: SøknadService,
+                                private val vedtakService: VedtakService,
+                                private val oppgaveService: OppgaveService,
+                                private val personService: PersonService
+) : AsyncTaskStep {
+    private val zoneIdOslo = ZoneId.of("Europe/Oslo")
+
+    override fun doTask(task: Task) {
+        val (behandlingId, hendelse, hendelseTidspunkt, gjeldendeSaksbehandler, oppgaveId) = objectMapper.readValue<BehandlingsstatistikkTaskPayload>(
+                task.payload
+        )
+
+        val personIdent = behandlingService.hentAktivIdent(behandlingId)
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        val fagsak = fagsakService.hentFagsak(behandling.fagsakId)
+
+        val sisteOppgaveForBehandling = finnSisteOppgaveForBehandlingen(behandlingId, oppgaveId)
+        val resultatBegrunnelse = finnResultatBegrunnelse(hendelse, behandlingId)
+        val søker = personService.hentSøker(personIdent);
```

Comment:
Burde vi hente data fra `GrunnlagsdataService` her? I stedet for å kalle på pdl?

Reply:
Jepp